cpp/include/cudf/detail/gather.cuh

```diff
@@ -45,6 +45,7 @@
 #include <thrust/logical.h>
 
 #include <cub/cub.cuh>
+#include <type_traits>
```

Comment:
The `<type_traits>` and `<cub/cub.cuh>` includes are not needed here as well.

Reply:
Done.